test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java

```diff
@@ -313,16 +314,18 @@ public void testCompactionInvalidTombstone() throws Throwable
     {
         DatabaseDescriptor.setCorruptedTombstoneStrategy(Config.CorruptedTombstoneStrategy.exception);
         prepare();
+
         // write a standard tombstone with negative local deletion time (LDTs are not set by user and should not be negative):
-        RowUpdateBuilder rub = new RowUpdateBuilder(getCurrentColumnFamilyStore().metadata(), -1, System.currentTimeMillis() * 1000, 22).clustering(33).delete("b");
-        rub.build().apply();
-        flush();
-        compactAndValidate();
-        readAndValidate(true);
-        readAndValidate(false);
+        Assertions.assertThatThrownBy(() -> {
+            new RowUpdateBuilder(getCurrentColumnFamilyStore().metadata(),
+                                 -1,
+                                 System.currentTimeMillis() * 1000,
+                                 22).clustering(33).delete("b");
+        }).isInstanceOf(IllegalArgumentException.class)
+          .hasMessageContaining("is outside the range for an unsigned integer value");
     }
 
-    @Test
+    @Test(expected = RuntimeException.class)  // msg = "is outside the range for an unsigned integer value"
```

Comment:
This means that now it fails on flush, so instead of generating a corrupt sstable we are not generating the sstable at all, right? Would all other writes for the same sstable be rejected too? Can this happen on a real use case?

By the way, the test could end in the call to `flush`

Reply:
This should never happen as negative ldts just can't get in now. The decision in [C14092](https://issues.apache.org/jira/browse/CASSANDRA-14092) was that you should scrub sstables if there were any negative ldts and _not_ support that logic in the storage engine.

So why does this test exist? digging a bit further [C14467](https://issues.apache.org/jira/browse/CASSANDRA-14467) seems to bring in some defensive logic towards negative ldts and other bad scenarios. I can only think this could happen on corrupted sstables which is what c14467 does: mark corrupted sstables. So I need to think if upon _decoding_ (bc we will never encode) a negative ldt outisde the valid uint range we should fail somehow we trigger the c14467 logic, wdyt?

Reply:
Maybe the solution would be along the lines as discussed [here](https://github.com/apache/cassandra/pull/1891#pullrequestreview-1297370063) which seems to fit nicely with the existing codebase if I am not missing anything.

Reply:
In the latest commit I generated some corrupted sstables, load them and trigger the sstable corruption detection path.